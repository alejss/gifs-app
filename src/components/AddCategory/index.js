import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ seetCategories }) => {
    const [inputValue, setinputValue] = useState('')

    const handleChangeText = (evn) => {
        const text = evn.target.value
        setinputValue(text)
    }

    const handleSubmit = (evn) => {
        evn.preventDefault()
        if (inputValue.trim().length > 2) seetCategories(categories => [inputValue, ...categories])
        setinputValue('')
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                placeholder='Add categories...'
                value={inputValue}
                onChange={handleChangeText}
            />
        </form>
    )
}

AddCategory.propTypes = {
    seetCategories: PropTypes.func.isRequired,
}
