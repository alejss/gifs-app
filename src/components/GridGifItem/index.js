import React from 'react'
import './style.css'

export const GridGifItem = ({ id, title, url }) => {
    return (
        <div className="card animate__animated animate__swing">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
