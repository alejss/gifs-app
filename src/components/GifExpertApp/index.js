import React, { useState } from 'react'
import './style.css'

// Components
import { AddCategory } from '../AddCategory'

export const GifExpertApp = () => {
  const [categories, seetCategories] = useState(['Cats', 'Dogs', 'Random'])
  return (
    <>
      <h2>Gif Expert App</h2>
      <hr />
      <AddCategory seetCategories={seetCategories} />
      <ol>
        {
          categories.map((categorie) => (<li key={categorie}>{categorie}</li>))
        }
      </ol>
    </>
  )
}
