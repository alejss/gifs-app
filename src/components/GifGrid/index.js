import React, { useState, useEffect } from 'react'
import './style.css'
import { GridGifItem } from '../GridGifItem'

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs()
    }, [])

    const getGifs = async () => {
        const url = `https://api.giphy.com/v1/gifs/search?q=Cats&limit=10&api_key=${process.env.REACT_APP_GIPHY_KEY}`
        const resp = await fetch(url)
        const { data } = await resp.json()

        const gifs = data.map(({ id, title, images }) => ({ id, title, url: images.downsized_medium.url }))
        setImages(gifs)
    }

    getGifs()

    return (
        <>
            <h3 key={category}>{category}</h3>
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GridGifItem
                            key={image.id}
                            {...image}
                        />)
                    )
                }
            </div>
        </>
    )
}
