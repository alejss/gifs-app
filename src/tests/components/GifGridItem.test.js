import { shallow } from 'enzyme'

import { GridGifItem } from '../../components/GridGifItem'

describe('GifGridItem', () => {
    const data = { title: 'Testing Component', url: 'www.1337.com/hello.png' }

    const wrapper = shallow(<GridGifItem {...data} />)

    test('Testing Snapshot Component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('The p tag must have a value with the title parameter', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(data.title)
    })

    test('the img tag must be equal to the url and alt of the props', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(data.url)
        expect(img.prop('alt')).toBe(data.title)
    })

    test('Contain class animate__animated animate__swing', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes('animate__animated animate__swing')).toBe(true)
    })
    
    
});