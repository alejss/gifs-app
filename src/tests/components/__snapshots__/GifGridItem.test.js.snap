// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem/> Testing Snapshot Component 1`] = `
<div
  className="card animate__animated animate__swing"
>
  <img
    alt="Testing Component"
    src="www.1337.com/hello.png"
  />
  <p>
    Testing Component
  </p>
</div>
`;
